histopath/parsers/geojson_parser.py

```diff
@@ -0,0 +1,105 @@
+import json
+from pathlib import Path
+from typing import Iterable
+
+import geopandas as gpd
+from geopandas import GeoDataFrame
+from shapely import Point, Polygon
+
+
+class GeoJSONParser:
+    """Parser for GeoJSON format annotation files.
+
+    GeoJSON is a format for encoding geographic data structures using JSON.
+    This parser supports both polygon and point geometries.
+    """
+
+    def __init__(self, file_path: Path | str) -> None:
+        self.file_path = Path(file_path)
+        self.gdf = gpd.read_file(self.file_path)
```

Comment:
```suggestion
    def __init__(self, file_path: Path | str) -> None:
        self.gdf = gpd.read_file(file_path)
```

The type should be also more general. Possibly the same as `gdp.read_file` input type